src/engine/HasPredicateScan.cpp

```diff
@@ -268,7 +272,7 @@ Result HasPredicateScan::computeResult([[maybe_unused]] bool requestLaziness) {
           TripleComponent::Iri::fromIriref(HAS_PATTERN_PREDICATE), std::nullopt,
           std::nullopt}
           .toScanSpecification(index);
-  const auto& perm = index.getPermutation(Permutation::Enum::PSO);
+  const auto& perm = permutation();
```

Comment:
It is doable to:

1. Create a proper index scan (member function call, because always the same), then call getResult(true) and in the generator, strip away the local vocab, because a range<IdTable> can be passed to the `computeFreeBla` functions.